tasks/utils.js

```diff
@@ -50,3 +51,39 @@ module.exports.spawnablePath = function (path) {
     }
     return path;
 };
+
+
+// return BrowserSync URL root
+var getBrowserSyncRootUrl = module.exports.getBrowserSyncRootUrl = function (options) {
+    return 'http://localhost:' + options.get('port');
+};
+
+// return BrowserSync JS client URL
+module.exports.getBrowserSyncClientUrl = function (options) {
+    var connectUtils = require('browser-sync/lib/connect-utils');
+    return getBrowserSyncRootUrl(options) + connectUtils.clientScript(options);
+};
+
+// Setup BrowserSync and catch errors gracefully
+module.exports.initBrowserSync = function (bs_opts, ready) {
+    var bs = require("browser-sync").create();
+    var opts = extend({
+        ui: false,
+        port: 35829,
+        ghostMode: false,
+        open: false,
+        notify: false,
+        logSnippet: false,
+        socket: {domain: getBrowserSyncRootUrl}
+    }, bs_opts);
+    var err;
+    try {
+        bs.init(opts, function (){
+            ready( err ? null : bs);
```

Comment:
I know this is unconventional, but it was the only way I could figure out how to get it to fail gracefully, because, for some reason, the callback IS being called regardless of an error being thrown. So, I am catching it below and storing the message in an in-scope 'err' variable to output it to the log before we continue (albeit without browserSync running). Try changing the port to 358299999. You will see an error message. BrowserSync will not start. But, the app will start anyway.